crates/turbopack/src/lib.rs

```diff
@@ -435,7 +435,7 @@ impl AssetContext for ModuleAssetContext {
                 request,
             );
 
-            result.add_reference(types_reference.into());
+            result.add_reference(types_reference.into()).await?;
```

Comment:
This probably returns a new result. Maybe rename the function to `with_reference`

```suggestion
            result = result.add_reference(types_reference.into());
```